src/js/RemotePlaybackPlugin.ts

```diff
@@ -0,0 +1,129 @@
+import type { RemotePlaybackState } from '../../@types/remote-playback';
+import type { VideoJsPlayer } from '../../@types/videojs';
+import { AirPlayManager, AirPlayButton } from './airplay/AirPlayManager';
+import videojs from '@silvermine/video.js';
+import { LOG_MESSAGES } from './/constants/log-messages';
+import { COMPONENT_NAMES } from './airplay/AirPlayButton';
+import { EVENTS } from './constants/remote-playback';
+
+const Plugin = videojs.getPlugin('plugin');
+
+export function isPlayerWithRemotePlaybackPlugin(o: unknown): o is VideoJsPlayer {
+   return typeof o === 'object' && o !== null &&
+      // Check for presence of some standard Player methods
+      'ready' in o && typeof o.ready === 'function' &&
+      'currentTime' in o && typeof o.currentTime === 'function' &&
+      'play' in o && typeof o.play === 'function' &&
+      'pause' in o && typeof o.pause === 'function' &&
+      // Finally, check for presence of our plugin
+      'remotePlayback' in o && typeof o.remotePlayback === 'function';
+}
+
+export function checkClientRemotePlaybackSupport(): boolean {
+   return typeof window !== 'undefined' &&
+      'HTMLVideoElement' in window &&
+      'remote' in HTMLVideoElement.prototype;
+}
+
+export class RemotePlaybackPlugin extends Plugin {
+
+   public readonly log!: videojs.Log;
+
+   private readonly _player: VideoJsPlayer;
+   private readonly _airPlayManager: AirPlayManager;
+
+   public constructor(player: videojs.Player) {
+      super(player);
+      // The constructor receives a standard Video.js Player, but then we assert the type
+      // of our custom VideoJsPlayer. This is safe because the plugin has been registered
+      // at this point.
+      this._player = player as VideoJsPlayer;
+      this._airPlayManager = new AirPlayManager(this._player);
+      this.log('initializing Remote Playback plugin...');
+
+      this._initialize();
+   }
+
+   public dispose(): void {
+      super.dispose();
+      this.log('Remote Playback plugin disposed');
+   }
+
+   public get airPlayManager(): AirPlayManager {
+      return this._airPlayManager;
+   }
+
+   public getState(): RemotePlaybackState | null {
+      return this._airPlayManager.getState();
+   }
+
+   public isConnected(): boolean {
+      return this._airPlayManager.isConnected();
+   }
+
+   private _initialize(): void {
+      this.log('Initializing Remote Playback plugin...');
+      this._addAirPlayButton();
+      this._checkAvailability();
+   }
+
+   private _addAirPlayButton(): void {
+      this.log('Initializing AirPlay button...');
+
+      this._player.ready(() => {
+         this.log(LOG_MESSAGES.PLAYER_READY);
+         this._addButtonToControlBar();
+      });
+   }
```

Comment:
Since `_addAirPlayButton` isn't actually adding the button, this is confusing.  How about if you just add the `ready` listener to part of `_initialize`?

```suggestion
   private _initialize(): void {
      this.log('Initializing Remote Playback plugin...');
      this._player.ready(() => {
         this.log(LOG_MESSAGES.PLAYER_READY);
         this._addButtonToControlBar();
      });
      this._checkAvailability();
   }
```